tests/gui/TestGuiFdoSecrets.cpp

```diff
@@ -761,6 +762,47 @@ void TestGuiFdoSecrets::testCollectionDelete()
     }
 }
 
+void TestGuiFdoSecrets::testHiddenFilename()
+{
+    // when file name contains leading dot, all parts excepting the last should be used
+    // for collection name, and the registration should success
+    QVERIFY(m_dbFile->rename(QFileInfo(*m_dbFile).path() + "/.Name.kdbx"));
+
+    m_db = nullptr;
```

Comment:
Must be `m_db.reset()`, technically unnecessary since you assign the pointer below anyway.

Reply:
Yeah noticed the build failure. It necessary though, otherwise the old database is not deleted and connections not removed until it's assigned below, at which point other things have changed and it's causing some problem.